bundles/org.openhab.binding.pjlinkdevice/src/main/java/org/openhab/binding/pjlinkdevice/internal/PJLinkDeviceHandler.java

```diff
@@ -0,0 +1,329 @@
+      } catch (AuthenticationException e) {
+        handleAuthenticationException(e);
+      } catch (ConfigurationException e) {
+        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, e.getMessage());
+      }
+    }, 0, TimeUnit.SECONDS);
+  }
+
+  protected PJLinkDeviceConfiguration getConfiguration() throws ConfigurationException {
+    PJLinkDeviceConfiguration config = this.config;
+    try {
+      validateConfigurationParameters(getThing().getConfiguration().getProperties());
+    } catch (ConfigValidationException e) {
+      String message = e.getValidationMessages().entrySet().stream()
+          .map((Map.Entry<String, String> a) -> (a.getKey() + ": " + a.getValue())).collect(Collectors.joining("; "));
+      throw new ConfigurationException(message);
+    }
+    if (config == null) {
+      this.config = config = getConfigAs(PJLinkDeviceConfiguration.class);
+    }
+    return config;
+  }
+
+  private void clearSetupJob() {
+    ScheduledFuture<?> setupJob = this.setupJob;
+    if (setupJob != null) {
+      setupJob.cancel(true);
+      this.setupJob = null;
+    }
+  }
+
+  private void clearRefreshInterval() {
+    ScheduledFuture<?> refreshJob = this.refreshJob;
+    if (refreshJob != null) {
+      refreshJob.cancel(true);
+      this.refreshJob = null;
+    }
+  }
+
+  private void handleAuthenticationException(AuthenticationException e) {
+    updateProperty(PJLinkDeviceBindingConstants.PROPERTY_AUTHENTICATION_REQUIRED, Boolean.TRUE.toString());
+    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, e.getMessage());
+  }
+
+  private void setupDevice() throws ConfigurationException, IOException, AuthenticationException, ResponseException {
+    PJLinkDevice device = getDevice();
+    device.checkAvailability();
+
+    updateDeviceProperties(device);
+    updateInputChannelStates(device);
+    updateStatus(ThingStatus.ONLINE);
+  }
+
+  private void setupRefreshInterval() throws ConfigurationException {
+    clearRefreshInterval();
+    PJLinkDeviceConfiguration config = PJLinkDeviceHandler.this.getConfiguration();
+    boolean atLeastOneChannelToBeRefreshed = config.refreshPower || config.refreshMute || config.refreshInputChannel;
+    if (config.refresh > 0 && atLeastOneChannelToBeRefreshed) {
+      refreshJob = scheduler.scheduleWithFixedDelay(() -> refresh(config), 0, config.refresh, TimeUnit.SECONDS);
+    }
+  }
+
+  private void updateDeviceProperties(PJLinkDevice device) throws IOException, AuthenticationException {
+    Map<String, String> properties = editProperties();
+
+    properties.put(PJLinkDeviceBindingConstants.PROPERTY_AUTHENTICATION_REQUIRED,
+        device.getAuthenticationRequired().toString());
+
+    try {
+      properties.put(PJLinkDeviceBindingConstants.PROPERTY_NAME, device.getName());
+    } catch (ResponseException e) {
+      logger.debug("Error retrieving property {}, cause: {}", PJLinkDeviceBindingConstants.PROPERTY_NAME, e);
```

Comment:
```suggestion
      logger.debug("Error retrieving property {}", PJLinkDeviceBindingConstants.PROPERTY_NAME, e);
```

stack trace will be printed anyway if added, also applies to other loggers

Reply:
You mean it will appear in the log even if I do not provide a placeholder? Could not find any documentation on that...

Reply:
https://www.baeldung.com/slf4j-log-exceptions